### PR TITLE
Support for setting NSMenuItem shortcut with nameless shortcut

### PR DESCRIPTION
**Pull request by pvieito**
- Support for setting a NSMenuItem with a nameless shortcut.
- Makes `Shortcut.keyEquivalent` public API.

### CODE REVIEW

**Comment by sindresorhus** [comment]
> Support for setting a NSMenuItem with a nameless shortcut.

What's the use-case?

**Comment by sindresorhus** [comment]
Look at your own diff and the warnings there.

**Comment by sindresorhus** [comment]
> Makes Shortcut.keyEquivalent public API.

Use-case?

File: Sources/KeyboardShortcuts/NSMenuItem++.swift [hunk 1]
```diff
@@ -31,31 +36,61 @@ extension NSMenuItem {
...
+	
+	This method overrides `.keyEquivalent` and `.keyEquivalentModifierMask`.
+	
+	- Important: You will have to remove the keyboard shortcut handlers while the menu is open, otherwise the keyboard events will be buffered up, the global handlers will be triggered when the menu closes and the events will not be correctly delivered to the menu. This is because `NSMenu` puts the thread in tracking-mode, which prevents the keyboard events from being received. You can listen to whether a menu is open by implementing `NSMenuDelegate#menuWillOpen` and `NSMenuDelegate#menuDidClose`, and use `KeyboardShortcuts.removeAllHandlers()` to remove the handlers when it is opened and recreate them once closed.
+	*/
+	public func setShortcut(_ shortcut: KeyboardShortcuts.Shortcut?) {
+		func clear() {
+			keyEquivalent = ""
+			keyEquivalentModifierMask = []
+		}
```
Comment (sindresorhus) [inline Sources/KeyboardShortcuts/NSMenuItem++.swift:82]: This is unsused.

File: Sources/KeyboardShortcuts/NSMenuItem++.swift [hunk 2]
```diff
@@ -31,31 +36,61 @@ extension NSMenuItem {
...
+	/**
+	Add a keyboard shortcut to a `NSMenuItem`.
+	
+	Pass in `nil` to clear the keyboard shortcut.
+	
+	This method overrides `.keyEquivalent` and `.keyEquivalentModifierMask`.
+	
+	- Important: You will have to remove the keyboard shortcut handlers while the menu is open, otherwise the keyboard events will be buffered up, the global handlers will be triggered when the menu closes and the events will not be correctly delivered to the menu. This is because `NSMenu` puts the thread in tracking-mode, which prevents the keyboard events from being received. You can listen to whether a menu is open by implementing `NSMenuDelegate#menuWillOpen` and `NSMenuDelegate#menuDidClose`, and use `KeyboardShortcuts.removeAllHandlers()` to remove the handlers when it is opened and recreate them once closed.
+	*/
+	public func setShortcut(_ shortcut: KeyboardShortcuts.Shortcut?) {
```
Comment (sindresorhus) [inline Sources/KeyboardShortcuts/NSMenuItem++.swift:78]:
```suggestion
	@_disfavoredOverload
	public func setShortcut(_ shortcut: KeyboardShortcuts.Shortcut?) {
```

**Comment by pvieito** [comment]
> > Support for setting a NSMenuItem with a nameless shortcut.
> 
> What's the use-case?

In my particular use case (the user can define shortcuts for items that are dynamically created), I would really prefer for the library to include nameless shortcut APIs everywhere (ie. `KeyboardShortcuts.onKeyDown(_ shortcut: Shortcut, action: @escaping KeyAction)`, `RecorderCocoa(onChange: (_ shortcut: Shortcut?) -> Void)? = nil)`), as the names are only an inconvenient addition as I already have to store them along user data.

File: Sources/KeyboardShortcuts/NSMenuItem++.swift (hunk 1 above)
Comment (pvieito) [reply Sources/KeyboardShortcuts/NSMenuItem++.swift]: Yup, removed, thanks!.

**Comment by pvieito** [comment]
> Look at your own diff and the warnings there.

Fixed!

**Comment by pvieito** [comment]
> > Makes Shortcut.keyEquivalent public API.
> 
> Use-case?

Removed as it is no longed required with new `NSMenuItem.setShortcut(_ shortcut: KeyboardShortcuts.Shortcut?)` API.

File: Sources/KeyboardShortcuts/NSMenuItem++.swift [hunk 3]
```diff
@@ -35,20 +40,46 @@ extension NSMenuItem {
...
+			else {
+				return
+			}
+
+			set()
+		}
+	}
+
+	/**
+	Add a keyboard shortcut to a `NSMenuItem`.
```
Comment (sindresorhus) [inline Sources/KeyboardShortcuts/NSMenuItem++.swift:70]: Can you mention that the other `setShortcut` method is the recommended one and that this one is useful for dynamic keyboard shortcut where you store the shortcuts yourself. Like, mention your use-case.

**Comment by sindresorhus** [comment]
And I'm happy to accept improvements to make it easier to use with dynamic shortcuts you store yourself. The current APIs are focused on what I needed myself.

File: Sources/KeyboardShortcuts/NSMenuItem++.swift (hunk 3 above)
Comment (pvieito) [reply Sources/KeyboardShortcuts/NSMenuItem++.swift:70]: Yes, of course. Done!

**Pull request merged**